### ISSUE TITLE
System.Buffers: XML doc in NuGet package (version 4.4.0 and later) differs from source code XML doc

### ISSUE THREAD

**Issue by stakx** [dotnet/corefx#28899]
**TL;DR:** Something appears to be wrong with the XML documentation included in the `System.Buffers` NuGet package, versions 4.4.0 and later. (4.4.0-preview2-25405-01 and earlier appear to have been fine*.) It doesn't match the XML documentation in the source code files.

----

I was reading the documentation for parameter `clearArray` of the method `ArrayPool<T>.Return(T[] array, bool clearArray)`. It goes as follows (formatting and emphasis by me):

> Indicates whether the contents of the buffer should be cleared before reuse. **If bufferLength is set to true** [...]. **If bufferLength is set to false** [...].

The reference to `bufferLength` doesn't make any sense. There's nothing public called `bufferLength` in this class. Here's a screenshot of the documentation showing the same text:

![XML documentation for `ArrayPool<T>.Return` as seen in Visual Studio](https://user-images.githubusercontent.com/104481/38443926-af5ca6f0-39ed-11e8-828b-8a725cacdc87.png)

Note that I'm using the latest package version, `4.5.0-preview1-26216-02`. The file `C:\Users\stakx\.nuget\packages\system.buffers\4.5.0-preview1-26216-02\version.txt` contains the hash `00490982c7952e96a1089818467caa6fd07541b2`, which I assume is a SHA-1 commit hash in this project's Git repository. Looking at that commit's version of `ArrayPool.cs`, this is what the documentation should read:

https://github.com/dotnet/corefx/blob/00490982c7952e96a1089818467caa6fd07541b2/src/System.Buffers/src/System/Buffers/ArrayPool.cs#L103-L108

No mention of `bufferLength` here, and this is the XML documentation you get with `System.Buffers` before version 4.4.0.

Looking at the version history of that file, there was never any `bufferLength` bit here as far as I can tell.

So what went wrong with the XML documentation starting with package version 4.4.0? Where does the `bufferLength` bit come from?

----

**(*)** XML docs in package versions before 4.4.0 were fine only in the sense that the documentation matched with that from the source code... but there were other, probably unrelated problems: The XML documentation file was either named incorrectly (just `.xml`) or missing (not in the `lib\` directory, but in the `ref\` directory), so one wouldn't get any documentation in VS IntelliSense.

**Comment by karelz** [comment]
I don't think NuGet XML documentation originates from CoreFX repo source code. @weshaggard?

**Comment by weshaggard** [comment]
That's correct the xml docs come from our docs repo @rpetrusha @mairaw

**Comment by mairaw** [comment]
Do you want to move this to dotnet/docs repo? We can investigate.

**Comment by stakx** [comment]
Can any one of you migrate this issue for me, or should I manually re-post over at dotnet/docs?

**Comment by karelz** [comment]
Here's the [doc page](https://docs.microsoft.com/en-us/dotnet/api/system.buffers.arraypool-1.return?view=netcore-2.0#System_Buffers_ArrayPool_1_Return__0___System_Boolean_).

Looking at history, it was already fixed in [this commit](https://github.com/dotnet/dotnet-api-docs/commit/b73fb6d7c45c7fe11d14c99c5eff0bcbb6ade13f#diff-b4cc564f29bf82db45a7915928c34864) on 2017/12. I assume it is just matter for new NuGet package to pick up the fresh documentation. @mairaw do you know if it happens automatically with new releases? (.NET Core 2.1)
I wonder if .NET Core 2.1 Preview1 already has the fix ...
